src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+import { CenterComponent, FormErrorText, StyledForm } from "@/components/Form/formStyling";
+import Button from "@mui/material/Button";
+import {
+    checkErrorOnSubmit,
+    Errors,
+    FormErrors,
+    setError,
+    setField,
+} from "@/components/Form/formFunctions";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBuildingCircleArrowRight } from "@fortawesome/free-solid-svg-icons";
+import RefreshPageButton from "@/app/admin/RefreshPageButton";
+import { Schema } from "@/database_utils";
+import NameCard from "@/app/admin/createCollectionCentre/createCollectionCentreFormSections/NameCard";
+import AcronymCard from "@/app/admin/createCollectionCentre/createCollectionCentreFormSections/AcronymCard";
+import supabase from "@/supabaseClient";
+
+const initialFields: Schema["collection_centres"] = {
+    name: "",
+    acronym: "",
+};
+
+const initialFormErrors: FormErrors = {
+    name: Errors.initial,
+    acronym: Errors.initial,
+};
+
+const formSections = [NameCard, AcronymCard];
+
+const CreateCollectionCentreForm: React.FC<{}> = () => {
+    const [fields, setFields] = useState(initialFields);
+    const [formErrors, setFormErrors] = useState(initialFormErrors);
+
+    const [submitError, setSubmitError] = useState(Errors.none);
+    const [submitErrorMessage, setSubmitErrorMessage] = useState("");
+    const [submitDisabled, setSubmitDisabled] = useState(false);
+
+    const [refreshRequired, setRefreshRequired] = useState(false);
```

Comment:
instead of this, directly call `router.refresh` or `window.location.reload`

Reply:
I ain't the person implementing this, but I think Daniel used this because this sets the submit button to a refresh button for people to click and refresh, instead of it just refreshing itself. Not exactly sure the reason behind, but I guess it's because there will be a confirmation shown from the button that the table has been updated successfully
So in this case i'd probably won't change it unless there is any specific issue with this implementation